torii/src/lib.rs

```diff
@@ -92,171 +95,171 @@ impl Torii {
-            });
+                }),
+            )
+            .route(
+                uri::CONFIGURATION,
+                post({
+                    let kiso = self.kiso.clone();
+                    move |Json(config): Json<_>| routing::handle_post_configuration(kiso, config)
+                }),
+            );
 
-        let blocks_ws_router = block_ws_router_path
-            .and(add_state!(self.kura))
-            .and(warp::ws())
-            .map(|sumeragi: Arc<_>, ws: Ws| {
-                ws.on_upgrade(|this_ws| async move {
-                    if let Err(error) = routing::handle_blocks_stream(sumeragi, this_ws).await {
-                        iroha_logger::error!(%error, "Failed to subscribe to blocks stream");
+        let router = router
+            .route(uri::SUBSCRIPTION, get({
+                let events = self.events.clone();
+                move |ws: WebSocketUpgrade| {
+                    core::future::ready(
+                        ws.on_upgrade(|ws| async move {
+                            if let Err(error) =
+                                routing::subscription::handle_subscription(events, ws).await
+                            {
+                                iroha_logger::error!(%error, "Failure during subscription");
+                            }
+                        })
+                    )
+                }
+            }))
+            .route(uri::BLOCKS_STREAM,
+                post({
+                    let kura = self.kura.clone();
+                    move |ws: WebSocketUpgrade| {
+                        core::future::ready(
+                            ws.on_upgrade(|ws| async move {
+                                if let Err(error) = routing::handle_blocks_stream(kura, ws).await {
+                                    iroha_logger::error!(%error, "Failed to subscribe to blocks stream");
+                                }
+                            })
+                        )
                     }
                 })
-            });
-
-        let ws_router = events_ws_router.or(blocks_ws_router);
-
-        warp::any()
-            .and(
-                // we want to avoid logging for the "health" endpoint.
-                // we have to place it **first** so that warp's trace will
-                // not log 404 if it doesn't find "/health" which might be placed
-                // **after** `.with(trace)`
-                health_route,
             )
-            .or(ws_router
-                .or(get_router)
-                .or(post_router)
-                .with(warp::trace::request()))
+            ;
+
+        router.layer((
+            TraceLayer::new_for_http()
+                .make_span_with(DefaultMakeSpan::default().include_headers(true)),
+            // Graceful shutdown will wait for outstanding requests to complete.
+            // Add a timeout so requests don't hang forever.
+            TimeoutLayer::new(Duration::from_secs(60)),
```

Comment:
Please define a `const`

```suggestion
            TimeoutLayer::new(SERVER_SHUTDOWN_TIMEOUT),
```